packages/core/test/buttons/buttonTests.tsx

```diff
@@ -84,5 +95,45 @@ function buttonTestSuite(component: React.ComponentClass<any>, tagName: string)
             const element = React.createElement(component, props);
             return useMount ? mount(element) : shallow(element);
         }
+
+        function checkClickTriggeredOnKeyUp(done: MochaDone,
+                                            buttonProps: Partial<IButtonProps>,
+                                            keyEventProps: Partial<React.KeyboardEvent<any>>) {
+            const wrapper = button(buttonProps, true);
+
+            // since Enzyme doesn't properly propagate events on non-string refs
+            // (https://github.com/airbnb/enzyme/issues/566), we need to dig in
+            // and attach a spy to the ref's .click function ourselves to ensure
+            // that it is being called. casting as `any` eliminates the button's
+            // knowledge of which members are public, private, or protected.
+            // this allows us to access the private buttonRef for the purposes
+            // of this test.
```

Comment:
this comment feels too verbose. should be enough to say "mock the DOM `click()` function because enzyme only handles simulated React events"